model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_horizontal_velocity_quantities.py

```diff
@@ -0,0 +1,269 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+import gt4py.next as gtx
+from gt4py.next.ffront.experimental import concat_where
+from gt4py.next.ffront.fbuiltins import astype, neighbor_sum
+
+from icon4py.model.atmosphere.dycore.stencils.accumulate_prep_adv_fields import (
+    _accumulate_prep_adv_fields,
+)
+from icon4py.model.atmosphere.dycore.stencils.compute_avg_vn import _compute_avg_vn
+from icon4py.model.atmosphere.dycore.stencils.compute_contravariant_correction import (
+    _compute_contravariant_correction,
+)
+from icon4py.model.atmosphere.dycore.stencils.compute_edge_diagnostics_for_velocity_advection import (
+    _compute_horizontal_kinetic_energy,
+    _interpolate_to_half_levels,
+)
+from icon4py.model.atmosphere.dycore.stencils.compute_mass_flux import (
+    _compute_mass_and_temperature_flux,
+)
+from icon4py.model.atmosphere.dycore.stencils.compute_tangential_wind import (
+    _compute_tangential_wind,
+)
+from icon4py.model.atmosphere.dycore.stencils.extrapolate_at_top import _extrapolate_at_top
+from icon4py.model.common import dimension as dims, field_type_aliases as fa, type_alias as ta
+from icon4py.model.common.dimension import E2C2EO, E2C2EODim
+from icon4py.model.common.type_alias import vpfloat
+
+
+@gtx.field_operator
+def _compute_horizontal_velocity_quantities_and_fluxes(
+    contravariant_correction_at_edges_on_model_levels: fa.EdgeKField[ta.vpfloat],
+    vn: fa.EdgeKField[ta.wpfloat],
+    e_flx_avg: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2C2EODim], ta.wpfloat],
+    geofac_grdiv: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2C2EODim], ta.wpfloat],
+    rbf_vec_coeff_e: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2C2EDim], ta.wpfloat],
+    rho_at_edges_on_model_levels: fa.EdgeKField[ta.wpfloat],
+    theta_v_at_edges_on_model_levels: fa.EdgeKField[ta.wpfloat],
+    ddqz_z_full_e: fa.EdgeKField[ta.vpfloat],
+    ddxn_z_full: fa.EdgeKField[ta.vpfloat],
+    ddxt_z_full: fa.EdgeKField[ta.vpfloat],
+    wgtfac_e: fa.EdgeKField[ta.vpfloat],
+    nflatlev: gtx.int32,
+) -> tuple[
+    fa.EdgeKField[ta.wpfloat],
+    fa.EdgeKField[ta.vpfloat],
+    fa.EdgeKField[ta.vpfloat],
+    fa.EdgeKField[ta.wpfloat],
+    fa.EdgeKField[ta.wpfloat],
+    fa.EdgeKField[ta.vpfloat],
+    fa.EdgeKField[ta.vpfloat],
+    fa.EdgeKField[ta.vpfloat],
+    fa.EdgeKField[ta.vpfloat],
+]:
+    spatially_averaged_vn = _compute_avg_vn(e_flx_avg=e_flx_avg, vn=vn)
+    horizontal_gradient_of_normal_wind_divergence = astype(
+        neighbor_sum(geofac_grdiv * vn(E2C2EO), axis=E2C2EODim), vpfloat
+    )
```

Comment:
it seems the functions are actually identical, but probably using the same as in 
```suggestion
    spatially_averaged_vn = _compute_avg_vn(e_flx_avg=e_flx_avg, vn=vn)
    horizontal_gradient_of_normal_wind_divergence = _compute_avg_vn(e_flx_avg=geofac_grdiv, vn=vn)
```
doesn't make sense (from a math point of view).

Reply:
But I guess either inline both or use a function for both. Below I have another suggestion for renaming `_compute_avg_vn` (therefore I probably prefer having functions).